sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/HttpResponse.java

```diff
@@ -88,10 +72,30 @@ public final HttpRequest getRequest() {
         return request;
     }
 
+    /**
+     * Buffers the {@link HttpResponse}.
+     * <p>
+     * If {@link #isBuffered()} is true, this instance of {@link HttpResponse} is returned. Otherwise, a new instance
+     * of {@link HttpResponse} is returned where {@link #getBody()} is buffered into memory.
+     *
+     * @return The buffered {@link HttpResponse}.
+     */
+    public abstract HttpResponse buffer();
+
+    /**
+     * Whether this {@link HttpResponse} is buffered.
+     *
+     * @return Whether this {@link HttpResponse} is buffered.
+     */
+    public abstract boolean isBuffered();
```

Comment:
same here, it seems a pure convenience and would always be the same as `getBody().isReplayable()` ?